### PR TITLE
book1: Fix include-guard issue with vec3.h listing

### LINKED ISSUE #1073: Syntax issue on vec3.h in book one
This goes back to #958, the `#endif` should appear at the bottom of the 2nd snippet.

Being where it currently is makes it yield code that will not compile if you just add the first and the second part. That means there's an error somewhere in the text.  The user could assume that the error is in the text "Listing 5: [vec3.h]" presuming that was meant to say "Listing 5: [vec3.cpp]" and thus still get invalid code.

Seeing a newline followed by an `#endif` is basically industry standard for meaning "This is the end of the file" so saying "The second part of the header file" doesn't really make any sense without some type of change.

screen

### PR DESCRIPTION
**Pull request by hollasch**
Resolves #1073

### CODE REVIEW

File: books/RayTracingInOneWeekend.html [hunk 1]
```diff
@@ -289,7 +289,7 @@
     using point3 = vec3;   // 3D point
     using color = vec3;    // RGB color
 
-    #endif
+    ...
```
Comment (trevordblack) [inline books/RayTracingInOneWeekend.html:292]: Keep the #endif where it is

File: books/RayTracingInOneWeekend.html [hunk 2]
```diff
@@ -305,6 +305,8 @@
 The second part of the header file contains vector utility functions:
 
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
+    ...
```
Comment (trevordblack) [inline books/RayTracingInOneWeekend.html:308]: Continue with a snippet from above

```
    ...
    
    // Type aliases for vec3
    using point3 = vec3;   // 3D point
    using color = vec3;    // RGB color
```

Comment (trevordblack) [reply books/RayTracingInOneWeekend.html:308]: Use highlighting on the changes

**Comment by hollasch** [comment]
Hmmm, generally we do not use highlighting when showing whole new sections added, otherwise the entire section is highlighted, which really isn't helpful.

**Comment by hollasch** [comment]
Actually, the whole "first part immediately followed by the second part" is somewhat clumsy. Perhaps I should just present the entire thing at once, and then adjust the text accordingly?

**Comment by hollasch** [comment]
Ok, took a second pass. I like this one a lot more.

**Review by trevordblack — APPROVED** [review]
Looks Good

**Pull request merged**